enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/ClientEvaluationService.java

```diff
@@ -0,0 +1,148 @@
+
+    /**
+     * All-arg constructor.
+     *
+     * @param onboardingProvider Onboarding provider.
+     * @param config Identity verification config.
+     * @param identityVerificationRepository Identity verification repository.
+     * @param transactionTemplate Transaction template.
+     */
+    @Autowired
+    public ClientEvaluationService(
+            final OnboardingProvider onboardingProvider,
+            final IdentityVerificationConfig config,
+            final IdentityVerificationRepository identityVerificationRepository,
+            final TransactionTemplate transactionTemplate) {
+        this.onboardingProvider = onboardingProvider;
+        this.config = config;
+        this.identityVerificationRepository = identityVerificationRepository;
+        this.transactionTemplate = transactionTemplate;
+    }
+
+    /**
+     * Process client evaluations.
+     */
+    @Transactional(readOnly = true)
+    public void processClientEvaluations() {
+        try (Stream<IdentityVerificationEntity> stream = identityVerificationRepository.streamAllInProgressClientEvaluations()) {
+            stream.forEach(this::processClientEvaluation);
+        }
+    }
+
+    private void processClientEvaluation(final IdentityVerificationEntity identityVerification) {
+        logger.debug("Evaluating client for {}", identityVerification);
+
+        final OwnerId ownerId = new OwnerId();
+        ownerId.setActivationId(identityVerification.getActivationId());
+        ownerId.setUserId("server-task-client-evaluations");
+
+        final EvaluateClientRequest request = EvaluateClientRequest.builder()
+                .processId(UUID.fromString(identityVerification.getProcessId()))
+                .userId(identityVerification.getUserId())
+                .identityVerificationId(identityVerification.getId())
+                .verificationId(getVerificationId(identityVerification))
+                .build();
+
+        final Consumer<EvaluateClientResponse> successConsumer = createSuccessConsumer(identityVerification);
+        final Consumer<Throwable> errorConsumer = createErrorConsumer(identityVerification);
+        final int maxFailedAttempts = config.getClientEvaluationMaxFailedAttempts();
+        onboardingProvider.evaluateClient(request)
+                .retryWhen(Retry.backoff(maxFailedAttempts, Duration.ofSeconds(2)))
+                .subscribe(successConsumer, errorConsumer);
+    }
+
+    private static String getVerificationId(final IdentityVerificationEntity identityVerification) {
+        return identityVerification.getDocumentVerifications().stream()
+                .findAny()
+                .map(DocumentVerificationEntity::getVerificationId)
+                .orElseThrow(() -> new IllegalStateException("No document verification for " + identityVerification));
+    }
+
+    private Consumer<EvaluateClientResponse> createSuccessConsumer(final IdentityVerificationEntity identityVerification) {
+        return response -> {
```

Comment:
Maybe we should use a separate service, e.g. new methods in `IdentityVerificationStatusService` to cover the status changes and rely on a standard transaction aware service method.

Reply:
You mean to add a save or update method to `IdentityVerificationService` or `IdentityVerificationStatusService` to avoid usage of `transactionTemplate`?

Reply:
Yeps, maybe separate those status changes to own method living in the `IdentityVerificationStatusService`. Currently those are just small snippets which could be more hard to test or discover.

Reply:
Ok, let's keep it this way, not to add dependency on another services. May be refactored later, if needed for tests.